### DISCUSSION TITLE
Upgrade wdio-allure-reporter to Allure 2

### DISCUSSION THREAD

**Discussion by hieuxlu** [webdriverio/webdriverio#6256] [Technical Discussion]
Allure 2 and its Javascript binding is currently on v2.0.0-beta.8, and under active development (https://github.com/allure-framework/allure-js) - last updated 20 days ago.

Upgrading will get rid of manually providing type definition for `allure-js-common`, because `allure-js` v2 uses Typescript (https://github.com/allure-framework/allure-js/tree/master/packages/allure-js-commons), just like wdio v7.

So I think this could be a good candidate to include in v7 of wdio and want to discuss about this. Thanks.

**Comment by christian-bromann** [comment]
@hieuxlu I am all for it. We will do the same for cucumber-js. They have release v7 yesterday that providings typings. We plan to release v7 by end of January. Would you be up to tackle this?

**Reply by hieuxlu** [reply]
I do. Now that wdio-allure.report is completely migrated to Typescript, we can start working on this.

**Reply by lacell75** [reply]
Have you plan in v8 the allure 2 with Cucumber.JS ? because allure-common-js is not maintained

**Reply by christian-bromann** [reply]
@lacell75 no but any help on this would be appreciated.
